### ISSUE TITLE
ACVP Demo Server reports a general exception - contact service provider

### ISSUE THREAD

**Issue by jbarnesweb** [usnistgov/ACVP-Server#82]
**environment**
Demo

**testSessionId**
166600

**vsId**
501636

**Algorithm registration**
RSA / keyGen / FIPS186-4

```

      {
        "algorithm": "RSA",
        "mode": "keyGen",
        "revision": "FIPS186-4",
        "prereqVals": [
          {
            "algorithm": "DRBG",
            "valValue": "same"
          },
          {
            "algorithm": "SHA",
            "valValue": "same"
          }
        ],
        "pubExpMode": "random",
        "keyFormat": "standard",
        "capabilities": [
          {
            "randPQ": "B.3.3",
            "properties": [
              {
                "modulo": 1024,
                "primeTest": [
                  "tblC2"
                ]
              },
              {
                "modulo": 2048,
                "primeTest": [
                  "tblC2"
                ]
              }
            ]
          }
        ]
      },

```

**Endpoint in which the error is experienced**
/acvp/v1/testSessions/166600/vectorSets/501636 GET

**Expected behavior**
download vector set 501636

**Comment by jbarnesweb** [comment]
Here's the http header returned

"headers\nHTTP/1.1 200 OK\r\nTransfer-Encoding: chunked\r\nContent-Type: application/json; charset=utf-8\r\nServer: Microsoft-IIS/8.5\r\nStrict-Transport-Security: max-age=31536000; includeSubDomains; preload\r\nDate: Wed, 03 Mar 2021 23:06:42 GMT\r\n\r\n"

Here's the content
"[{\"acvVersion\":\"1.0\"},{\"error\":\"General exception. Contact service provider.\"}]"

**Comment by jbarnesweb** [comment]
It would make it easier to handle on the client if the server didn't return http status 200 for errors like this.

**Comment by Kritner** [comment]
Thanks @jbarnesweb, this should have been caught and rejected when attempting to do the test session post, rather than on the get prompt attempt.

The modulo 1024 is not a valid modulo for RSA KeyGen 186-4, we should have rejected the registration up front.  I'll get in a change to do so in the next release.  

Regarding returning a different status code in instances of such a failure, I think we have an issue or two out there for that, but we won't be looking at making any protocol changes until after the issue backlog gets more cleaned up.

FIPS186-4 *does* specify that 1024 modulo is valid when utilize with non random prime generation, although there is https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-131Ar2.pdf:

![image](https://user-images.githubusercontent.com/8814983/116723863-ec127100-a9ad-11eb-8a27-4be8a916bfdf.png)

which supersede FIPS 186-4, and disallows modulos < 2048

**Comment by Kritner** [comment]
This change is on demo in release [v1.1.0.18](https://github.com/usnistgov/ACVP-Server/releases/tag/v1.1.0.18)

**Comment by Kritner** [comment]
This change has been released to prod in release [v1.1.0.18](https://github.com/usnistgov/ACVP-Server/releases/tag/v1.1.0.18).
